text/0000-conditional-compilation-checking.md

```diff
@@ -0,0 +1,515 @@
+condition name as both enabled and valid. For example, the following invocations are equivalent:
+
+### The `values(...)` form
+
+The `values(...)` form enables checking the values within list-valued conditions. It has this
+form:
+
+```bash
+rustc --check-cfg `values(name, "value1", "value2", ... "valueN")'
+```
+
+where `name` is a bare identifier (has no quotes) and each `"value"` term is a quoted literal
+string. `name` specifies the name of the condition, such as `feature` or `target_os`.
+
+When the `values(...)` option is specified, `rustc` will check every `#[cfg(name = "value")]`
+attribute, `#[cfg_attr(name = "value")]` attribute, and `cfg!(name = "value")` call. It will
+check that the `"value"` specified is present in the list of valid values. If `"value"` is not
+valid, then `rustc` will report an `invalid_cfg_name` lint diagnostic. The default diagnostic
+level for this lint is `Warn`.
+
+The form `values()` is an error, because it does not specify a condition name.
+
+To enable checking of values, but to provide an empty set of valid values, use this form:
+
+```bash
+rustc --check-cfg `values(name)`
+```
+
+The `--check-cfg values(...)` option can be repeated, both for the same condition name and for
+different names. If it is repeated for the same condition name, then the sets of values for that
+condition are merged together.
+
+> The `--check-cfg names(...)` and `--check-cfg values(...)` options are independent. `names`
+> checks the namespace of condition names; `values` checks the namespace of the values of
+> list-valued conditions.
+
+### Valid values can be split across multiple options
+
+The valid condition values are the union of all options specified on the command line.
+For example, this command line:
+
+```bash
+# legal but redundant:
+rustc --check-cfg 'values(animals, "lion")' --check-cfg 'values(animals, "zebra")'
+
+# equivalent:
+rustc --check-cfg 'values(animals, "lion", "zebra")'
+```
+
+This is intended to give tool developers more flexibility when generating Rustc command lines.
+
+### Enabled conditions are implicitly valid
+
+Specifying an enabled condition name implicitly makes it valid. For example, the following
+invocations are equivalent:
+
+```bash
+# legal but redundant:
+rustc --check-cfg 'values(animals, "lion", "zebra")' --cfg 'animals = "lion"'
+
+# equivalent:
+rustc --check-cfg 'values(animals, "zebra")' --cfg 'animals = "lion"'
+```
+
+Specifying an enabled condition _value_ implicitly makes that _value_ valid. For example, the
+following invocations are equivalent:
+
+```bash
+# legal but redundant
+rustc --check-cfg 'values(animals, "lion", "zebra")' --cfg 'animals = "lion"'
+
+# equivalent
+rustc --check-cfg 'values(animals, "zebra")' --cfg 'animals = "lion"'
+```
+
+Specifying a condition value also implicitly marks that condition _name_ as valid. For example,
+the following invocations are equivalent:

```

Comment:
Very minor, but it may be worth noting that this is talking about specifying the condition name and value in a `--check-cfg` argument, where as the previous two examples are about specifying the name and/or value in a `--cfg` argument.

Reply:
Yeah, I hope this text / example gets that subtlety across. If we think there's risk of confusion, I suppose we could expand on that.